te2rules/adapter.py

```diff
@@ -229,3 +233,121 @@ def _convert(self) -> DecisionTree:
 
         root_node = nodes[0]
         return root_node
+
+
+class XgboostXGBClassifierAdapter:
+    """
+    Class to convert xgboost.sklearn.XGBClassifier
+    into a te2rules.tree.RandomForest object.
+
+    Usage:
+    adapter = XgboostXGBClassifierAdapter(model, feature_names)
+    adapted_model = adapter.random_forest
+    """
+
+    def __init__(self, xgb_model: XGBClassifier, feature_names: List[str]):
+        self.xgb_model = xgb_model
+        self.feature_names = feature_names
+
+        # TODO: n0 and n1 are not accessible from the xgboost model itself, in this way bias term will be 0
+        self.n0 = 1
+        self.n1 = 1
+        self.bias = np.log(self.n1 / self.n0)
+
+        # Default learning rate 0.3
+        self.weight = self.xgb_model.get_params()["learning_rate"] or 0.3
```

Comment:
Since xgboost doesn't use a separate weight variable, `self.weight` can be set to 1. scikit is slighlty different in this aspect. It uses learning rate as tree weights, while xgboost does not.